src/sql-parser/src/ast/mod.rs

```diff
@@ -258,6 +261,10 @@ pub enum Expr {
     Nested(Box<Expr>),
     /// A literal value, such as string, number, date or NULL
     Value(Value),
+    TypedString {
```

Comment:
No strong feelings from me but seems like we could also just parse this as a cast and avoid adding a new AST node?

Reply:
Yeah I went back and forth on that a lot. Definitely can parse as a cast, but it seemed like maybe `SHOW CREATE VIEW` would want to preserve `SELECT date '2010-01-01'` rather than rewriting to `SELECT CAST ('2010-01-01' as date)`. Hmm, maybe we could instead rewrite to `SELECT '2010-01-01'::date`. What do you think?

Reply:
I like `'2010-01-01'::date`! In general I'm sympathetic to both lowering the number of AST nodes and to not mangling the input too much at the parsing phase, so I'm happy either way.

Reply:
👍 went with the cast now that the output for casts is more succinct.